### PR TITLE
Fix integration tests.

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
- remove the need for having external config and mail.properties files to run integration tests
- fixed `ContextRegionAutoConfiguration` and `ContextCredentialsAutoConfiguration` - beans have to be registered in bean definition phase
- improved tests execution speed by replacing explicit waits with Awaitility
- improved tests resilience 
- added JUnit 5 compatible `IfAmazonWebserviceEnvironmentCondition`

### CODE REVIEW

**Comment by maciejwalkowiak** [comment]
Apologies @eddumelendez for large PR.

File: README.adoc [hunk 1]
```diff
@@ -86,36 +81,14 @@ itself. The last password rdsPassword is used to access the database
...
-
--------------------------------------------
-cloud.aws.credentials.accessKey=ilaugsjdlkahgsdlaksdhg
-cloud.aws.credentials.secretKey=aöksjdhöadjs,höalsdhjköalsdjhasd+
-rdsPassword=someVerySecretPassword
--------------------------------------------
-
-Also you have to create another file named mail.properties which will
-provide the sender and recipient mail address to test the
+Also you have to prepare the sender and recipient mail addresses to test the
```
Comment (eddumelendez) [inline README.adoc:84]:
```suggestion
Also you must prepare the sender and recipient mail addresses to test the
```

File: docs/src/main/asciidoc/README.adoc [hunk 2]
```diff
@@ -80,36 +75,14 @@ itself. The last password rdsPassword is used to access the database
...
-
--------------------------------------------
-cloud.aws.credentials.accessKey=ilaugsjdlkahgsdlaksdhg
-cloud.aws.credentials.secretKey=aöksjdhöadjs,höalsdhjköalsdjhasd+
-rdsPassword=someVerySecretPassword
--------------------------------------------
-
-Also you have to create another file named mail.properties which will
-provide the sender and recipient mail address to test the
+Also you have to prepare the sender and recipient mail addresses to test the
```
Comment (eddumelendez) [inline docs/src/main/asciidoc/README.adoc:78]:
```suggestion
Also you must prepare the sender and recipient mail addresses to test the
```

File: spring-cloud-aws-integration-test/src/test/java/org/springframework/cloud/aws/it/mail/BootMailSenderAwsTest.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
```
Comment (eddumelendez) [inline spring-cloud-aws-integration-test/src/test/java/org/springframework/cloud/aws/it/mail/BootMailSenderAwsTest.java:2]:
```suggestion
 * Copyright 2013-2020 the original author or authors.
```

File: spring-cloud-aws-integration-test/src/test/java/org/springframework/cloud/aws/it/support/profile/IfAmazonWebserviceEnvironmentCondition.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
```
Comment (eddumelendez) [inline spring-cloud-aws-integration-test/src/test/java/org/springframework/cloud/aws/it/support/profile/IfAmazonWebserviceEnvironmentCondition.java:2]:
```suggestion
 * Copyright 2002-2020 the original author or authors.
```

**Pull request merged**